modules/common/src/main/java/org/opencastproject/mediapackage/MediaPackageBuilderImpl.java

```diff
@@ -100,6 +108,25 @@ public MediaPackage createNew(Id identifier) {
    * @see org.opencastproject.mediapackage.MediaPackageBuilder#loadFromXml(java.io.InputStream)
    */
   public MediaPackage loadFromXml(InputStream is) throws MediaPackageException {
+
+    if (serializer != null) {
+      try {
+
+        //Convert InputStream to XML document to rewrite the URLs
+        Document xml = XmlSafeParser.parse(is);
+        rewriteUrls(xml, serializer);
+
+        // Reconverting back to inputStream
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        Source xmlSource = new DOMSource(xml);
+        Result outputTarget = new StreamResult(os);
+        XmlSafeParser.newTransformerFactory().newTransformer().transform(xmlSource, outputTarget);
+        is = new ByteArrayInputStream(os.toByteArray());
```

Comment:
@mliradelc it's because your solution is calling `rewriteUrls`, and @lkiesow's does not.

Try this, starting from the same point as the above suggestion:
```
        //Convert InputStream to XML document to rewrite the URLs
        Document xml = XmlSafeParser.parse(is);
        rewriteUrls(xml, serializer);
        return loadFromXml(xml);
```